xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -98,37 +99,116 @@ public class RunSync {
                   + "used for any Iceberg source or target.")
           .addOption(HELP_OPTION, "help", false, "Displays help information to run this utility");
 
-  public static void main(String[] args) throws IOException {
-    CommandLineParser parser = new DefaultParser();
+  static SourceTable sourceTableBuilder(
+      DatasetConfig.Table table,
+      IcebergCatalogConfig icebergCatalogConfig,
+      DatasetConfig datasetConfig,
+      Properties sourceProperties) {
+    Objects.requireNonNull(table, "Table cannot be null");
+    Objects.requireNonNull(datasetConfig, "datasetConfig cannot be null");
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(table.getTableName())
+            .basePath(table.getTableBasePath())
+            .namespace(table.getNamespace() == null ? null : table.getNamespace().split("\\."))
+            .dataPath(table.getTableDataPath())
+            .catalogConfig(icebergCatalogConfig)
+            .additionalProperties(sourceProperties)
+            .formatName(datasetConfig.sourceFormat)
+            .build();
+    return sourceTable;
+  }
 
-    CommandLine cmd;
-    try {
-      cmd = parser.parse(OPTIONS, args);
-    } catch (ParseException e) {
-      new HelpFormatter().printHelp("xtable.jar", OPTIONS, true);
-      return;
-    }
+  static List<TargetTable> targetTableBuilder(
+      DatasetConfig.Table table,
+      IcebergCatalogConfig icebergCatalogConfig,
+      List<String> tableFormatList) {
+    Objects.requireNonNull(table, "Table cannot be null");
+    Objects.requireNonNull(tableFormatList, "tableFormatList cannot be null");
+    List<TargetTable> targetTables =
+        tableFormatList.stream()
+            .map(
+                tableFormat ->
+                    TargetTable.builder()
+                        .name(table.getTableName())
+                        .basePath(table.getTableBasePath())
+                        .namespace(
+                            table.getNamespace() == null ? null : table.getNamespace().split("\\."))
+                        .catalogConfig(icebergCatalogConfig)
+                        .formatName(tableFormat)
+                        .build())
+            .collect(Collectors.toList());
+    return targetTables;
+  }
 
-    if (cmd.hasOption(HELP_OPTION)) {
-      HelpFormatter formatter = new HelpFormatter();
-      formatter.printHelp("RunSync", OPTIONS);
-      return;
+  static void formatConvertor(
+      DatasetConfig datasetConfig,
+      List<String> tableFormatList,
+      IcebergCatalogConfig icebergCatalogConfig,
```

Comment:
`org.apache.xtable.conversion.CatalogConfig`
Can you refactor this to ? It's better to use interfaces instead of the implementation class for common methods like these which are format agnostic. 

icebergCatalogConfig -> catalogConfig

Reply:
Sorry, I couldn't quite get it what I have to change. @vinishjail97 
Could you please explain more.

Reply:
do you mean changing the` IcebergCatalogConfig icebergCatalogConfig = getIcebergCatalogConfig(icebergCatalogConfigpath);`
`CatalogConfig icebergCatalogConfig = getIcebergCatalogConfig(icebergCatalogConfigpath);`
and using icebergCatalogConfig of type  `CatalogConfig` further in code? @vinishjail97

Reply:
IcebergCatalogConfig is one of the implementation for the interface, the arguments to these functions having the interface allows other implementation classes to be passed in. 
```
  static SourceTable sourceTableBuilder(
      @NonNull DatasetConfig.Table table,
      CatalogConfig catalogConfig,
      @NonNull DatasetConfig datasetConfig,
      Properties sourceProperties) {  
```

Reply:
@vinishjail97 I have done the required changes in the functions. Please review.